src/components/textbox/textbox.test.tsx

```diff
@@ -0,0 +1,607 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import * as floatingUi from "@floating-ui/react-dom";
+import Textbox, { TextboxProps } from ".";
+import { testStyledSystemMargin } from "../../__spec_helper__/__internal__/test-utils";
+import { EnterKeyHintTypes } from "../../__internal__/input";
+import FormFieldStyle from "../../__internal__/form-field/form-field.style";
+import createGuid from "../../__internal__/utils/helpers/guid";
+import CarbonProvider from "../carbon-provider/carbon-provider.component";
+import Logger from "../../__internal__/utils/logger";
+import StyledInput from "../../__internal__/input/input.style";
+
+jest.mock("../../__internal__/utils/logger");
+
+const mockedGuid = "mocked-guid";
+jest.mock("../../__internal__/utils/helpers/guid");
+
+(createGuid as jest.MockedFunction<typeof createGuid>).mockReturnValue(
+  mockedGuid
+);
+
+let loggerSpy: jest.SpyInstance<void, [message: string]> | jest.Mock;
+
+beforeEach(() => {
+  loggerSpy = jest.spyOn(Logger, "deprecate");
+});
+
+afterEach(() => {
+  loggerSpy.mockRestore();
+});
+
+afterAll(() => {
+  loggerSpy.mockClear();
+});
+
+test("should display deprecation warning once when rendered as uncontrolled", () => {
+  render(<Textbox name="my-textbox" defaultValue="test" />);
+
+  expect(loggerSpy).toHaveBeenCalledWith(
+    "Uncontrolled behaviour in `Textbox` is deprecated and support will soon be removed. Please make sure all your inputs are controlled."
+  );
+
+  expect(loggerSpy).toHaveBeenCalledTimes(1);
+});
+
+testStyledSystemMargin(
+  (props) => <Textbox {...props} />,
+  undefined,
+  (component) => component.find(FormFieldStyle),
+  { modifier: "&&&" }
+);
+
+describe(`when the characterLimit prop is passed`, () => {
+  test.each([2, 3, 4])("renders a character counter", (characterLimit) => {
```

Comment:
suggestion(non-blocking): if a child of a describe `it` should be used instead of `test`

Describe blocks combined with it blocks should read as continuous prose to describe the test. Converting the above (and the other `test` blocks) to an `it` would work

Reply:
ah OK, thanks for this - I'd looked at other PRs and thought that a decision had been reached to change all `it`'s to `test` so had just been blindly doing this everywhere!

Reply:
I get what @nineteen88 is saying, I'm not overly fussed with what we use inside the describes tbh, it makes more sense to follow the `describe -> it` pattern but I don't think the tests are unreadable or the same meaning can't be gathered from them using `describe ->  test` so happy to defer to you on whether you update